sdk/nodejs/provider/server.ts

```diff
@@ -272,33 +284,8 @@ class Server implements grpc.UntypedServiceImplementation {
     }
 
     public async construct(call: any, callback: any): Promise<void> {
-        // Serialize invocations of `construct` and `call` so that each call runs one after another, avoiding concurrent
-        // runs. We do this because `construct` and `call` modify global state to reset the SDK's runtime options.
-        // This is a short-term workaround to provide correctness, but likely isn't sustainable long-term due to the
-        // limits it places on parallelism. We will likely want to investigate if it's possible to run each invocation
-        // in its own context, possibly using Node's `createContext` API to avoid modifying global state:
-        // https://nodejs.org/api/vm.html#vm_vm_createcontext_contextobject_options
-        const res = this.constructCallQueue.then(() => this.constructImpl(call, callback));
-        /* eslint-disable no-empty,no-empty-function,@typescript-eslint/no-empty-function */
-        this.constructCallQueue = res.catch(() => {});
-        return res;
-    }
-
-    async constructImpl(call: any, callback: any): Promise<void> {
-        // given that construct calls are serialized, we can attach an uncaught handler to pick up exceptions
-        // in underlying user code. When we catch the error, we need to respond to the gRPC request with the error
-        // to avoid a hang.
-        const uncaughtHandler = (err: Error) => {
-            if (!this.uncaughtErrors.has(err)) {
-                this.uncaughtErrors.add(err);
-            }
-            // bubble the uncaught error in the user code back and terminate the outstanding gRPC request.
-            callback(err, undefined);
-        };
-        process.on("uncaughtException", uncaughtHandler);
-        // @ts-ignore 'unhandledRejection' will almost always invoke uncaughtHandler with an Error. so
-        // just suppress the TS strictness here.
-        process.on("unhandledRejection", uncaughtHandler);
+        const callbackId = Symbol("id");
+        this._callbacks.set(callbackId.toString(), callback);
```

Comment:
I need to check but I feel like you could use the `Symbol` directly as a `Map` key, so `Map<Symbol, ...>`? But I could be wrong/there could be a good reason not to do this.

Reply:
Ah you're right. Wasn't aware of that!